File: allinone-kotlin/app/src/main/java/com/example/allinone/data/HistoryItem.kt
package com.example.allinone.data

import java.util.Date

/**
 * Represents an item in the history list.
 * This is a unified representation of different data types.
 */
data class HistoryItem(
    val id: Long,
    val title: String,
    val description: String,
    val date: Date,
    val amount: Double? = null,
    val type: String,
    val imageUri: String? = null,
    val itemType: ItemType
) {
    enum class ItemType {
        TRANSACTION,
        TRANSACTION_INCOME,
        TRANSACTION_EXPENSE,
        INVESTMENT,
        NOTE,
        REGISTRATION
    }
} 

File: src/features/history/types/HistoryItem.ts
export type HistoryItemType = 'TRANSACTION_INCOME' | 'TRANSACTION_EXPENSE' | 'INVESTMENT' | 'REGISTRATION';

export interface HistoryItem {
  id: string;
  title: string;
  description: string;
  date: string; // ISO string
  amount?: number;
  currency?: 'TRY' | 'AED' | 'USD';
  type: string;
  imageUri?: string;
  itemType: HistoryItemType;
}
